### PR TITLE
Allow WIP marker to contains < or >

### LINKED ISSUE #4705: If the first word at WORK_IN_PROGRESS_PREFIXES starts with <, the hint is shown incorrectly
- Gitea version (or commit ref): 03e558c
(the rest are not relevant or difficult)

## Description

Since #4529 a commit can be marked as _WIP_. The _WIP_ prefix can be configured. We have marked our PR with a `<WIP>` prefix format since a while, so we put that as the default. I have this in my app.ini:

```ini
[repository.pull-request]
; List of prefixes used in Pull Request title to mark them as Work In Progress
WORK_IN_PROGRESS_PREFIXES=<WIP>,WIP:,[WIP]
```

It works pretty well, except for the hint text that doesn't show the prefix to use:

image

You can see the hint only says `Start the title with`, without the `<WIP>`. For reference, here is how is showed with default values:

image

After a little of experimentation, it seems the problem occurs any time the prefix start with a `<` character.

@JulienTant

### PR DESCRIPTION
**Pull request by JulienTant**
Fixes #4705 

When WIP prefix contained  `< >`, then the helper link didn't show the prefix because it was parsed by HTML. This PR provides the view an html escaped version of the first prefix (=the default one for UI)

### CODE REVIEW

File: routers/repo/issue.go [hunk 1]
```diff
@@ -356,6 +357,7 @@ func NewIssue(ctx *context.Context) {
 	ctx.Data["RequireHighlightJS"] = true
 	ctx.Data["RequireSimpleMDE"] = true
 	ctx.Data["RequireTribute"] = true
+	ctx.Data["FirstPullRequestWorkInProgressPrefix"] = html.EscapeString(setting.Repository.PullRequest.WorkInProgressPrefixes[0])
```
Comment (lafriks) [inline routers/repo/issue.go]: Escaping should be done in template

**Comment by lafriks** [comment]
You can use: `(.FirstPullRequestWorkInProgressPrefix | Escape) | Safe`

**Comment by JulienTant** [comment]
> You can use: `(.FirstPullRequestWorkInProgressPrefix | Escape) | Safe`

I didn't know about the `Escape`, and copied the `| Safe` from somewhere else... I searched in the godoc (html/template) but found nothing about those modifiers/functions :/ If you have a link to provide me with more info about templating I would be so happy :)

**Comment by lafriks** [comment]
`Escape` is our own template helper. For others non standard template helpers used in Gitea check file `modules/templates/helper.go`

**Comment by lafriks** [comment]
btw what was reason for creating that new template context variable?
I'm not 100% sure but probably this will also work:
`((index .PullRequestWorkInProgressPrefixes 0) | Escape) | Safe`

**Comment by techknowlogick** [comment]
> btw what was reason for creating that new template context variable?

@lafriks I think because `WIP` might be different in other languages and so this allows users to modify to suit their needs. @JulienTant please correct me if I am incorrect.

**Comment by lafriks** [comment]
@techknowlogick yes but with index it would also work I think same as it was before, just with additional encoding

**Comment by techknowlogick** [comment]
@lafriks oh sorry. I thought you were talking about why have the variable in general, not specifically how the variable was used. I understand now and agree with your suggestion as above.

**Comment by JulienTant** [comment]
Hi, the goal was to escape the first option but yeah this is not needed now

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
make LG-TM work

**Review by techknowlogick — APPROVED** [review]

**Review by tboerger — APPROVED** [review]

**Pull request merged**